test/unit/reset-ui.test.js

```diff
@@ -1,21 +1,81 @@
 /* eslint-env qunit */
 import TestHelpers from './test-helpers.js';
+import {createTimeRanges} from '../../src/js/utils/time.js';
+import sinon from 'sinon';
 QUnit.module('player reset-ui');
 
-QUnit.test('Calling resetProgressBar player method should place progress bar at 0% width', function(assert) {
-  const player = TestHelpers.makePlayer();
+QUnit.test('Calling resetProgressBar should reset the components displaying time related values', function(assert) {
+  const clock = sinon.useFakeTimers();
+  const player = TestHelpers.makePlayer({controls: true});
 
+  clock.tick(1000);
+  player.trigger('ready');
+  player.buffered = function() {
+    return createTimeRanges(0, 5);
+  };
+  player.duration(40);
   player.currentTime(20);
+
+  const {
+    currentTimeDisplay,
+    durationDisplay,
+    remainingTimeDisplay,
+    progressControl: {
+      seekBar
+    }
+  } = player.controlBar;
+
+  clock.tick(30);
+
   player.trigger('timeupdate');
+  player.trigger('progress');
+
+  clock.tick(30);
+
+  // Current time display
+  assert.equal(currentTimeDisplay.textNode_.textContent, '0:20', 'current time display is 0:20');
+  // Duration display
+  assert.equal(durationDisplay.textNode_.textContent, '0:40', 'duration display is 0:40');
+  // Remaining time display
+  assert.equal(remainingTimeDisplay.textNode_.textContent, '0:20', 'remaining time display is 0:20');
+  // Seek bar
+  assert.equal(seekBar.getProgress(), '0.5', 'seek bar progress is 0.5');
+  assert.equal(seekBar.getAttribute('aria-valuetext'), '0:20 of 0:40', 'seek bar progress holder aria value text is 0:20 of 0:40');
+  assert.equal(seekBar.getAttribute('aria-valuenow'), '50.00', 'seek bar progress holder aria value now is 50.00');
+  // Load progress
+  assert.equal(seekBar.loadProgressBar.el().textContent, 'Loaded: 12.50%', 'load progress bar textContent is Loaded: 12.50%');
+  assert.equal(seekBar.loadProgressBar.el().style.width, '12.5%', 'load progress bar width is 12.5%');
+  // Play progress
+  assert.equal(seekBar.playProgressBar.el().textContent, '0:20', 'player progress bar textContent is 0:20');
+  assert.equal(seekBar.playProgressBar.el().style.width, '50%', 'player progress bar width is 50%');
+  assert.equal(seekBar.playProgressBar.timeTooltip.el().textContent, '0:20', 'player progress bar time tooltip is 0:20');
+
+  // Do reset
   player.resetProgressBar_();
-  assert.equal(
-    player.controlBar.progressControl.seekBar.playProgressBar.el().offsetWidth, 0,
-    'progress bar is reset to width 0%'
-  );
-  assert.equal(
-    player.currentTime(), 0,
-    'player current time is 0'
-  );
+  player.duration(0);
```

Comment:
question: is there a reason that duration is set to 0 here? 

I think it would make more sense if the player had a duration of more than 0  so that the tests show `resetProgressBar_` works with a duration of more than 0.

Reply:
@usmanonazim thank you for your feedback 👍🏽 

> question: is there a reason that duration is set to 0 here?

Yes, the main reason is that we test directly the function `resetProgressBar_`. The time reset is done by the function `reset`, that will eventually call [`Html5.resetMediaElement`](https://github.com/videojs/video.js/blob/2a99a7824ea175836249a4044c1d530cce174e08/src/js/tech/html5.js#L1356-L1382) which will cause a reset of `durationDisplay`. This means that in this test case the time must be reset manually otherwise the duration will be `40` see:

https://github.com/amtins/video.js/blob/9d58ef7d8b9bb445b38d6f626b7c613d602a730e/test/unit/reset-ui.test.js#L16

> I think it would make more sense if the player had a duration of more than 0 so that the tests show resetProgressBar_ works with a duration of more than 0.

I'm not quite sure I understand. The `resetProgressBar_` function calling the `durationDisplay.updateContent` if the duration value is not zero then the component will not be reset. 
https://github.com/videojs/video.js/blob/2a99a7824ea175836249a4044c1d530cce174e08/src/js/control-bar/time-controls/duration-display.js#L69-L74

Reply:
Ahh I see thank you for the explanation! Ignore the second part of my comment I was getting a little confused with the way the reset worked.